jenkins_pipelines/environments/common/pipeline-build-validation-cleanup.groovy

```diff
@@ -0,0 +1,249 @@
+def run(params) {
+    timestamps {
+        // Define paths and environment variables for reusability
+        String SUSEManagerCleanerProgram = "./susemanager-ci/jenkins_pipelines/scripts/SUSEManager_cleaner/suse_manager_cleaner_program/SUSEManagerCleaner.py"
+        GString resultdir = "${WORKSPACE}/results"
+        GString resultdirbuild = "${resultdir}/${BUILD_NUMBER}"
+        GString exports = "export BUILD_NUMBER=${BUILD_NUMBER}; export BUILD_VALIDATION=true; "
+        String container_repository = params.container_repository ?: null
+        String product_version = null
+        String controllerHostname = null
+        String serverHostname = null
+        GString targetedTfFile = "/home/jenkins/workspace/${params.targeted_project}/results/sumaform/main.tf"
+        GString targetedTfStateFile = "/home/jenkins/workspace/${params.targeted_project}/results/sumaform/terraform.tfstate"
+        GString targetedTerraformDirPath = "/home/jenkins/workspace/${params.targeted_project}/results/sumaform/"
+        GString localSumaformDirPath = "${resultdir}/sumaform/"
+        GString localTfStateFile = "${localSumaformDirPath}/terraform.tfstate"
+        GString logFile = "${resultdirbuild}/sumaform.log"
+
+        // Construct the --tf-resources-to-delete argument dynamically
+        ArrayList defaultResourcesToDelete = []
+        if (params.clean_proxy) {
+            defaultResourcesToDelete.add('proxy')
+        }
+        if (params.clean_monitoring_server) {
+            defaultResourcesToDelete.add('monitoring-server')
+        }
+        if (params.clean_retail) {
+            defaultResourcesToDelete.add('retail')
```

Comment:
Actually, if we add retail, we already wrap proxy, so we don't really need to add proxy on such case.

Reply:
That is true. It's a bit hard to have this modularity. I think we may simplify it later with two options: client cleanup or full cleanup. It's hard to be in-between without making mistakes. I will not address this change now but create a card. https://github.com/SUSE/spacewalk/issues/26010